tests/hybig/test_hybig.py

```
@@ -0,0 +1,105 @@
+
+
+def test_hybig(failed_tests, harmony_client, service_collection):
+    """Run a request against HyBIG and make sure it is successful.
+
+    As a lightweight example, this test will check the Harmony request
+    returned a successful status and the output STAC contains all expected
+    files. No outputs will be downloaded for further verification to minimise
+    overall runtime of the test suite.
+
+    Test fixtures are retrieved from `tests/conftest.py`, which contains
+    fixtures common to all Harmony services under test.
+
+    """
+    harmony_request = Request(
+        collection=Collection(id=service_collection['concept_id']),
+        max_results=1,
+        format='image/png',
+    )
+
+    try:
+        # Submit the job and get the JSON output once completed
+        harmony_job_id = harmony_client.submit(harmony_request)
+        result_json = harmony_client.result_json(harmony_job_id)
+
+        # Check the response was successful
+        assert result_json['status'] == 'successful', (
+            f'Harmony request failed: {result_json["message"]}'
+        )
+
+        # Check the URLs for results are all of the expected type.
+        ensure_correct_files_created(result_json['links'])
+    except AssertionError as exception:
+        # Cache error message and re-raise the AssertionError to fail the test
+        failed_tests.append(
+            {
+                **service_collection,
+                'error': str(exception),
+            }
+        )
+        raise
+    except Exception as exception:
+        # Catch other exception types and raise as an AssertionError to
+        # ensure test test suite is robust against unexpected exceptions.
+        # This does not cache the failure, as this should only arise from
+        # systematic issues, such as connecting to Harmony, not issues specific
+        # to the collection under test.
+        raise AssertionError('Unexpected request failure') from exception
```

is this the right error to raise here? `RuntimeError` maybe?

---

My thinking was that the `AssertionError` would let `pytest` have a chance to run all the other tests. Otherwise, I think the thing to do would be to not catch non-`AssertionError` exceptions and let them just be raised and halt the test suite.

I'm on the fence. This feels a tiny bit more graceful, because it plays nicer with `pytest`, but getting to this point of the code is likely due to something that will fail for everything. Maybe not catching non-`AssertionError` exceptions at all letting the tests fall over is better than trying to run through a bunch more tests that are going to fail.

---

Ok, that sort of make sense why you kept it as assertion error. 
So you're failing the test, but not writing it in the output? is that all?

---

Yeah - exactly that.